### PR TITLE
Fix(web, web-twig, web-react): Tooltip on Icon component

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

Added forwardedRef to our `<Icon />` component so it can be used as `elementType` for `<TooltipTrigger>` component.

Added new demo for tooltip on Icon components for all packages

image

### Additional context

Requested in our support channel:
https://almamedia.slack.com/archives/C068XPSDWQN/p1706807163184249

### CODE REVIEW

**Comment by dlouhak** [comment]
Thanks for those changes! 🙏🏻 

Could you also update [documentation](https://github.com/lmc-eu/spirit-design-system/tree/main/packages/web-react/src/components/Tooltip#tooltipmodern)? We're little bit missing mention about `TooltipTrigger` or `TooltipPopover` API.

File: packages/web-twig/src/Resources/components/Tooltip/stories/TooltipIcon.twig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          UNSAFE_className="is-hidden"
+          enableControlledPlacement
+          placement="right"
+          flipFallbackPlacements="top-start, top-bottom, left"
+          isDismissible
+      >
+          Close me
+      </TooltipPopov>
+  </TooltipWrapper>
+</span>
```
Comment (literat) [inline packages/web-twig/src/Resources/components/Tooltip/stories/TooltipIcon.twig:24]: No new line at the end?

**Review by literat — APPROVED** [review]
LGTM 👍

**Comment by pavelklibani** [comment]
moved to https://github.com/lmc-eu/spirit-design-system/pull/1268

**Pull request closed without merging**